transcom/mymove
# B 20940 int 2

## [B-20940](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20940)

> [!NOTE]  
> This is a follow up PR to https://github.com/transcom/mymove/pull/13574 , created in order to resolve some unexpected behavior from the originally supplied calculation

## Summary
During PO Acceptance, our PO had to change a couple requirements about the calculation.

Originally it was planned that Member Disbursement should not be less than 0, however in the case that a member has requested an advance greater than their expenses, this value should be shown as negative to reflect the difference that they owe back to the government.

For this solution, it’s important to note that GTCC disbursement will only ever be less than 0 using the calculation of (remaining entitlement + member paid storage). In this case, member disbursement will be set to the value of the negative GTCC disbursement. GTCC disbursement will be set to 0. The other calculation of (GTCC Totaled other expenses + GTCC Storage expense) can never be less than 0.

In all other cases, the calculation will perform as outlined in the AC. This results in the disbursement numbers we are looking for.

This change has been described to the PO, and they agree that it's the best way forward.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the application as a service member. 
2. Create a PPM with a comically large Advance request.
3. When adding information about the move in terms of expenses, keep them to a minimum. The idea is to create a negative remaining PPM incentive.
4. Log in as a service counselor and keep the move going between the two roles to reach the closeout process. 
5. As a customer in the closeout process, create a very small member-paid storage expense. This will help us ensure things are calculated properly. 
6. Complete the closeout process and download the Payment Packet. 
7. Ensure GTCC Disbursement is 0.
8. Ensure member disbursement is equal to the negative remaining PPM incentive, plus the positive member-paid storage expense, resulting in a closer-to-zero negative number. This is the amount you now owe the government :)

### Backend
